ml/Config.cc

```diff
@@ -57,6 +62,32 @@ void Config::readMLConfig(void) {
     SS << netdata_configured_cache_dir << "/anomaly-detection.db";
     Cfg.AnomalyDBPath = SS.str();
 
+    #if defined(ENABLE_ML_TESTS)
+    std::stringstream TestDBDirectory, TestDataDirectory, TestQuery1Directory, TestCheck1Directory, TestQuery2Directory, TestCheck2Directory;
+    TestDBDirectory << netdata_configured_cache_dir << "/ml_test_anomaly_info.db";
+    AnomalyTestDBPath = TestDBDirectory.str();
+    Cfg.AnomalyTestDBPath = AnomalyTestDBPath;
+
+    TestDataDirectory << netdata_configured_cache_dir << "/ml_test_data.sql";
+    AnomalyTestDataPath = TestDataDirectory.str();
+    Cfg.AnomalyTestDataPath = AnomalyTestDataPath;
+    
+    TestQuery1Directory << netdata_configured_cache_dir << "/ml_test_query_1.sql";
+    AnomalyTestQuery1Path = TestQuery1Directory.str();
+    Cfg.AnomalyTestQuery1Path = AnomalyTestQuery1Path;
+
+    TestCheck1Directory << netdata_configured_cache_dir << "/ml_test_check_1.sql";
+    AnomalyTestCheck1Path = TestCheck1Directory.str();
+    Cfg.AnomalyTestCheck1Path = AnomalyTestCheck1Path;
+
+    TestQuery2Directory << netdata_configured_cache_dir << "/ml_test_query_2.sql";
+    AnomalyTestQuery2Path = TestQuery2Directory.str();
+    Cfg.AnomalyTestQuery2Path = AnomalyTestQuery2Path;
+
+    TestCheck2Directory << netdata_configured_cache_dir << "/ml_test_check_2.sql";
+    AnomalyTestCheck2Path = TestCheck2Directory.str();
+    Cfg.AnomalyTestCheck2Path = AnomalyTestCheck2Path;
+    #endif // ENABLE_ML_TESTS
```

Comment:
I think we need to structure tests a little bit better in order to avoid mixing business logic with test logic. Let's execute and verify tests with `googletest`.

It's fine if you assume that tests will manually run under the source repository in order to avoid creating all these variables.

> `TestDBDirectory << netdata_configured_cache_dir << "/ml_test_anomaly_info.db";`

I think that just for testing we can use a memory only DB, right?

Reply:
Agree with the separate test code files, but couldn't make sure if memory would not be hindered for the longer run profiling test.

Reply:
Tests now using googletest.